valhalla/baldr/directededge.h

```diff
@@ -1157,7 +1172,8 @@ class DirectedEdge {
   uint64_t bss_connection_ : 1; // Does this lead to(come out from) a bike share station?
   uint64_t stop_sign_ : 1;      // Stop sign at end of the directed edge
   uint64_t yield_sign_ : 1;     // Yield/give way sign at end of the directed edge
-  uint64_t spare4_ : 7;
+  uint64_t z_level_ : 4;        // Z-level of the edge
```

Comment:
i dont think this should go in the directededge, it makes more sense in edgeinfo. first because its a property of the edge that doesnt change based on the direction of travel and second because we wont be using it to change the costing function for the edge. its a "display" property, those belong in edgeinfo. now the next problem is that edgeinfo is already chaulk full and its harder to modify hahaha but i do believe it belongs there

Reply:
Well :) 
What do you think if we add [1 bit here](https://github.com/valhalla/valhalla/blob/79c404631c475607411252b78de1ee97a0467ed0/valhalla/baldr/edgeinfo.h#L207) indicating if we have smth like `EdgeInfoInnerExt` written after `EdgeInfoInner`? `EdgeInfoInnerExt` could look smth like:
```
struct EdgeInfoInnerExt {
   uint64_t z_level: 4;
   uint64_t z_level: 60;
}
```
?

Reply:
Or we could come up with some more sophisticated variable-size encoding...

Reply:
Probably we could utilize these things here https://en.wikipedia.org/wiki/Elias_delta_coding or https://en.wikipedia.org/wiki/Elias_gamma_coding. It would allow us to not waste bits when they are not actually needed for the cost of additional bits for each value 🤔 
In other words at the moment we have to have this `EdgeInfoInnerExt` of fixed size and z-level is always 4 bit. For example with delta coding we could encode our [-6;8] range with 1-8 bits depending on actual value and do not waste bits if they are unused.
WDYT?